drivers/dma/dma_ti_cc23x0.c

```diff
@@ -0,0 +1,380 @@
+		return -EINVAL;
+	}
+
+	if (config->dma_slot > DMA_CC23_IPID_MASK) {
+		LOG_ERR("Invalid trigger");
+		return -EINVAL;
+	}
+
+	if (config->block_count > 1) {
+		LOG_ERR("Chained transfers not supported");
+		return -ENOTSUP;
+	}
+
+	switch (config->source_data_size) {
+	case 1:
+		src_inc_flags = UDMA_SRC_INC_8;
+		break;
+	case 2:
+		src_inc_flags = UDMA_SRC_INC_16;
+		break;
+	case 4:
+		src_inc_flags = UDMA_SRC_INC_32;
+		break;
+	default:
+		LOG_ERR("Invalid source data size");
+		return -EINVAL;
+	}
+
+	switch (config->dest_data_size) {
+	case 1:
+		dst_inc_flags = UDMA_DST_INC_8;
+		break;
+	case 2:
+		dst_inc_flags = UDMA_DST_INC_16;
+		break;
+	case 4:
+		dst_inc_flags = UDMA_DST_INC_32;
+		break;
+	default:
+		LOG_ERR("Invalid destination data size");
+		return -EINVAL;
+	}
+
+	data_size = MIN(config->source_data_size, config->dest_data_size);
+
+	switch (data_size) {
+	case 1:
+		control = UDMA_SIZE_8;
+		break;
+	case 2:
+		control = UDMA_SIZE_16;
+		break;
+	case 4:
+		control = UDMA_SIZE_32;
+		break;
+	default:
+		LOG_ERR("Invalid data size");
+		return -EINVAL;
+	}
+
+	ret = dma_cc23x0_set_addr_adj(&control, block->source_addr_adj,
+				      src_inc_flags, UDMA_SRC_INC_NONE);
+	if (ret) {
+		LOG_ERR("Invalid source address adjustment type");
+		return ret;
+	}
+
+	ret = dma_cc23x0_set_addr_adj(&control, block->dest_addr_adj,
+				      dst_inc_flags, UDMA_DST_INC_NONE);
+	if (ret) {
+		LOG_ERR("Invalid dest address adjustment type");
+		return ret;
+	}
+
+	xfer_size = block->block_size / data_size;
+	if (!xfer_size || xfer_size > UDMA_XFER_SIZE_MAX) {
+		LOG_ERR("Invalid block size (must be in range %d to %d)",
+			data_size, data_size * UDMA_XFER_SIZE_MAX);
+		return -EINVAL;
+	}
+
+	burst_len = config->source_burst_length / data_size;
+	if (burst_len <= UDMA_XFER_SIZE_MAX && IS_POWER_OF_TWO(burst_len)) {
+		control |= LOG2(burst_len) << UDMA_ARB_S;
+	} else {
+		LOG_ERR("Invalid burst length (must be a power of 2 between %d and %d)",
+			data_size, data_size * UDMA_XFER_SIZE_MAX);
+		return -EINVAL;
+	}
+
+	ch_data = &data->channels[channel];
+	ch_data->data_size = data_size;
+	ch_data->cb = config->dma_callback;
+	ch_data->user_data = config->user_data;
+
+	if (uDMAIsChannelEnabled(BIT(channel))) {
+		return -EBUSY;
+	}
+
+	if (DMA_CC23_IS_SW_CH(channel)) {
+		uDMAEnableSwEventInt(BIT(channel));
+	} else {
+		/* Select peripheral */
+		DMA_CC23_CHXSEL_REG(channel) = config->dma_slot;
+	}
+
+	uDMASetChannelControl(&data->desc[channel], control);
+
+	uDMASetChannelTransfer(&data->desc[channel], DMA_CC23_MODE(channel),
+			       (void *)block->source_address,
+			       (void *)block->dest_address,
+			       xfer_size);
+
+	LOG_DBG("Configured channel %d for %08x to %08x (%u bytes)",
+		channel,
+		block->source_address,
+		block->dest_address,
+		block->block_size);
+
+	return 0;
+}
+
+static int dma_cc23x0_start(const struct device *dev, uint32_t channel)
+{
+	if (uDMAIsChannelEnabled(BIT(channel))) {
```

Comment:
Repeated start shouldn't result in a failure, if the channel is already going then start should be viewed as a NOP